### DISCUSSION TITLE
Go to

### DISCUSSION THREAD

**Discussion by davidmaoconnor** [okaybenji/text-engine#15] [Q&A] [answered]
Does anyone know how I can incorporate a "go to" function? I want to be able to "go to" a room instead of using cardinal directions.

**Comment by okaybenji** [comment, marked as answer]
Hey, sorry I did not see this until now! Please feel free to tag me @okaybenji with questions.

There's a function you can call to move a player to a room called `enterRoom`. You just need to pass it the ID of the room to go to.

In order to allow the player to go to any room, you would need to either override the existing `go` command, or you could just add a `goto` command that lives alongside it. This is the simplest, as you can just use `enterRoom` directly.

Here's a simple example. Your player starts in the first room.

```
// Basic game disk with two rooms.
const disk = {
  roomId: 'start',
  rooms: [
    { id: 'start', name: 'The First Room' },
    { id: 'end, name: 'The Last Room' },
  ]
};

// Override commands that accept a single argument to include custom GOTO command.
commands[1] = Object.assign(commands[1], {goto: enterRoom});
```

Now, if your player enters "goto end", they will be in the last room.

Let me know if you have any questions about this.
